content/blog/analysis-with-RCrunch.md

```diff
@@ -0,0 +1,504 @@
+    list(expression=so_data$Professional == "Student" &
+             so_data$ProgramHobby %in% c("Yes, I contribute to open source projects", "Yes, both"),
+         name="Student open source contributor"),
+    list(expression=so_data$ProgramHobby %in% c("No", "Yes, I program as a hobby"),
+         name="Open source non-contributor"))
+
+so_data[['OScontributor']] <- makeCaseVariable(cases=case_statements,
+    name = "Open source contribution (professional/student)")
+
+so_data[['OScontributor']]
+```
+
+    ## Open source contribution (professional/student) (categorical)
+    ##
+    ##                                      Count
+    ## Open source non-contributor          34588
+    ## Professional open source contributor 12264
+    ## Student open source contributor       2362
+
+`as.data.frame` is helpful for connecting with functions that want a dataframe
+------------------------------------------------------------------------------
+
+To get an even more data-frame like object, `as.data.frame` will produce a *Crunch dataframe* object which acts more like a standard R dataframe, but still doesn't pull all of the data from the Crunch server. This is useful for functions that require a dataframe `data` argument (like `lm`). The special part of *Crunch dataframes* is that although they act like regular dataframes (e.g. you get R vectors when you access its columns), the data for each column is only retrieved when it is needed.
+
+Using `as.data.frame`, we can fit a linear regression with a single predictor of `HomeRemote` predicting `JobSatisfaction`. The syntax for `lm` is exactly the same as it would be to use a local dataframe. But because we are using a *Crunch dataframe*, only the data that `lm` needs is transferred and only when R needs it.
+
+``` r
+so_df <- as.data.frame(so_data)
+fit <- lm(JobSatisfaction~HomeRemote, data=so_df)
+summary(fit)
+```
+
+    ##
+    ## Call:
+    ## lm(formula = JobSatisfaction ~ HomeRemote, data = so_df)
+    ##
+    ## Residuals:
+    ##     Min      1Q  Median      3Q     Max
+    ## -7.3884 -1.0775  0.3675  1.3675  3.3675
+    ##
+    ## Coefficients:
+    ##                                                                   Estimate Std. Error t value Pr(>|t|)
+    ## (Intercept)                                                        6.63250    0.01918 345.760  < 2e-16 ***
+    ## HomeRemoteA few days each month                                    0.44501    0.02614  17.027  < 2e-16 ***
+    ## HomeRemoteLess than half the time, but at least one day each week  0.48762    0.03966  12.294  < 2e-16 ***
+    ## HomeRemoteAbout half the time                                      0.34120    0.05910   5.773 7.85e-09 ***
+    ## HomeRemoteMore than half, but not all, the time                    0.32068    0.05711   5.616 1.97e-08 ***
+    ## HomeRemoteAll or almost all the time (I'm full-time remote)        0.75593    0.03784  19.976  < 2e-16 ***
+    ## ---
```

Comment:
I would be remiss if I didn’t suggest `arm::display` or coefplot?
And that simulations are always better than tables of coefs… maybe a ggplotly graph?